src/test/java/org/datadog/jmeter/plugins/DatadogBackendClientTest.java

```diff
@@ -169,12 +172,18 @@ public void testExtractMetrics() {
     }
 
     @Test
-    public void testExtractLogs() {
+    public void testExtractLogs() throws IOException{
         SampleResult result = createDummySampleResult("foo");
         this.client.handleSampleResults(Collections.singletonList(result), context);
         Assert.assertEquals(1, this.logsBuffer.size());
         String expectedPayload = "{\"sample_start_time\":1.0,\"response_code\":\"123\",\"headers_size\":0.0,\"sample_label\":\"foo\",\"latency\":12.0,\"group_threads\":0.0,\"idle_time\":0.0,\"error_count\":0.0,\"message\":\"\",\"url\":\"\",\"ddsource\":\"jmeter\",\"sent_bytes\":124.0,\"thread_group\":\"bar\",\"body_size\":0.0,\"content_type\":\"\",\"load_time\":125.0,\"thread_name\":\"bar baz\",\"sample_end_time\":126.0,\"bytes\":12345.0,\"connect_time\":0.0,\"sample_count\":10.0,\"data_type\":\"\",\"all_threads\":0.0,\"data_encoding\":null}";
-        Assert.assertEquals(this.logsBuffer.get(0).toString(), expectedPayload);
+        JSONParser parser = new JSONParser();
+        try {
+            Assert.assertEquals(parser.parse(this.logsBuffer.get(0).toString()), parser.parse(expectedPayload));
+        }
+        catch (ParseException ex) {
+            throw new IOException("Error while parsing payload from testExtractLogs ", ex);
+        }
     }
```

Comment:
What do you think about this solution? 

```java
@Test
public void testExtractLogs() throws ParseException {
    SampleResult result = createDummySampleResult("foo");
    this.client.handleSampleResults(Collections.singletonList(result), context);
    Assert.assertEquals(1, this.logsBuffer.size());
    String expectedPayload = "{\"sample_start_time\"1.0,\"response_code\":\"123\",\"headers_size\":0.0,\"sample_label\":\"foo\",\"latency\":12.0,\"group_threads\":0.0,\"idle_time\":0.0,\"error_count\":0.0,\"message\":\"\",\"url\":\"\",\"ddsource\":\"jmeter\",\"sent_bytes\":124.0,\"thread_group\":\"bar\",\"body_size\":0.0,\"content_type\":\"\",\"load_time\":125.0,\"thread_name\":\"bar baz\",\"sample_end_time\":126.0,\"bytes\":12345.0,\"connect_time\":0.0,\"sample_count\":10.0,\"data_type\":\"\",\"all_threads\":0.0,\"data_encoding\":null}";
    JSONParser parser = new JSONParser(JSONParser.MODE_PERMISSIVE);
    Assert.assertEquals(this.logsBuffer.get(0), parser.parse(expectedPayload));
}
```

Note that `new JSONParser()` is deprecated.

Also I think the `ParseException` dos not need to get wrapped, it is already clear enough, example: 

```
net.minidev.json.parser.ParseException: Unexpected character (1) at position 19.
```

Reply:
Yes I think it's better, I'll make the changes accordingly. Thanks for the advice.

Reply:
Updated. @FlorentClarret